src/Interpreters/Cache/QueryCache.cpp

```diff
@@ -142,9 +142,7 @@ bool QueryCache::Key::operator==(const Key & other) const
 String QueryCache::Key::queryStringFromAst() const
 {
     WriteBufferFromOwnString buf;
-    IAST::FormatSettings format_settings(buf, /*one_line*/ true);
-    format_settings.show_secrets = false;
-    ast->format(format_settings);
+    ast->format(buf, IAST::FormatSettings{.show_secrets = false});
```

Comment:
Can't we write just `ast->format(buf, {.show_secrets = false});` ? Because it's shorter

Reply:
Fixed.

However, it was a conscious decision to spell out `IAST::FormatSettings` so that it's more obvious to the reader, and the usages of `IAST::FormatSettings` are more searchable.

I do like your option better for readability. I guess, people can just look up `format` signature, and they could search for all `format` entries instead.